org.eclipse.xtext.util/src/org/eclipse/xtext/util/LineAndColumn.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2015, 2019 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.util;
+
+import java.util.Objects;
+
+import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
+
+/**
+ * A simple record with line and column information. The line number is
+ * one-based, e.g. the first line in the document has the line number 1. The
+ * column number is one based, too, e.g the first column in a line has the index
+ * 1.
+ * 
+ * Line breaks are part of the line itself, e.g. the first line break in the
+ * document still has line number one.
+ * 
+ * @author Sebastian Zarnekow - Initial contribution and API
+ * @since 2.9
+ */
+public class LineAndColumn {
+	private final int line;
+
+	private final int column;
+
+	public static LineAndColumn from(int line, int column) {
+		LineAndColumn result = new LineAndColumn(line, column);
+		if (line <= 0 || column <= 0) {
+			throw new IllegalArgumentException(result.toString());
+		}
+		return result;
+	}
+
+	private LineAndColumn(int line, int column) {
+		this.line = line;
+		this.column = column;
+	}
+
+	public int getLine() {
+		return line;
+	}
+
+	public int getColumn() {
+		return column;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(line, column);
```

Comment:
Allocates an array on each invocation -I’m not a big fan

Reply:
Also: boxing if I’m not mistaken

Reply:
I see, thanks for this tip.